generate/newlicense.go

```diff
@@ -28,6 +33,66 @@ import (
 	"github.com/gravitational/trace"
 )
 
+func newCertificate(licenseParams NewLicenseInfo) ([]byte, error) {
+	ca, err := licenseParams.CACertificate()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	signingKey, err := licenseParams.SigningKey()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	loopbackIP := net.ParseIP(constants.LoopbackIP)
+	if loopbackIP == nil {
+		return nil, trace.BadParameter("loopback IP is invalid (this is a bug)")
+	}
+
+	// Note: starting with Go 1.24, the serial number is generated automatically.
+	// Keep this here for now until dependent repos are all on Go 1.24.
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return nil, trace.Wrap(err, "generating serial number")
+	}
+
+	template := x509.Certificate{
+		Subject: pkix.Name{
+			CommonName:   constants.LicenseKeyPair,
+			Organization: []string{constants.LicenseOrg},
+		},
+		SerialNumber:       serialNumber,
+		NotBefore:          time.Now().Add(-5 * time.Minute),
+		NotAfter:           time.Now().Add(licenseParams.ValidFor),
+		IPAddresses:        []net.IP{loopbackIP},
+		KeyUsage:           x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
+		ExtKeyUsage:        []x509.ExtKeyUsage{x509.ExtKeyUsageServerAuth, x509.ExtKeyUsageClientAuth},
+		SignatureAlgorithm: x509.SHA256WithRSA,
+		ExtraExtensions: []pkix.Extension{
+			{
+				Id:    constants.LicenseASN1ExtensionID,
+				Value: licenseParams.Payload, // TODO: hex encode?
```

Comment:
The hex-encoding seems to just be a cfssl-ism, the payload is not encoded in our license files.

```suggestion
				Value: licenseParams.Payload,
```